### PR TITLE
Add dialog for repeating password when exporting keys

### PR DESCRIPTION
**Pull request by Integral-Tech**
Related issue: https://github.com/Nheko-Reborn/nheko/issues/1729

### CODE REVIEW

**Comment by uwnuk** [comment]
I suggest considering whether to have it compare passwords upon every character entry.
Maybe it's not the best idea - I do understand that nheko aims to be fast, and this could be more complicated, however it can offer a better user experience in some cases.

File: src/UserSettingsPage.cpp [hunk 1]
```diff
@@ -2127,11 +2127,25 @@ UserSettingsModel::exportSessionKeys()
...
     if (password.isEmpty()) {
         QMessageBox::warning(nullptr, tr("Error"), tr("The password cannot be empty"));
         return;
     }
 
+    if (password != repeatedPassword) {
+        QMessageBox::warning(nullptr, tr("Error"), tr("Passwords don't match"));
+        return;
+    }
+
```
Comment (uwnuk) [inline src/UserSettingsPage.cpp:2148]: We may want to warn that the repeated password is empty if they've entered a password.
This should be obvious to the end user anyways, but
```suggestion
    if (password.isEmpty()) {
        QMessageBox::warning(nullptr, tr("Error"), tr("The password cannot be empty"));
        return;
    }

    if (repeatedPassword.isEmpty()) {
        QMessageBox::warning(nullptr, tr("Error"), tr("Please repeat the password"));
        return;
    }

    if (password != repeatedPassword) {
        QMessageBox::warning(nullptr, tr("Error"), tr("Passwords don't match"));
        return;
    }

```

**Comment by Integral-Tech** [comment]
@uwnuk Thanks for your suggestion

**Comment by Integral-Tech** [comment]
@deepbluev7

File: src/UserSettingsPage.cpp [hunk 2]
```diff
@@ -2127,11 +2127,30 @@ UserSettingsModel::exportSessionKeys()
...
+                                                  QLineEdit::Password,
+                                                  QLatin1String(""),
+                                                  &ok);
+    if (!ok)
+        return;
+
     if (password.isEmpty()) {
         QMessageBox::warning(nullptr, tr("Error"), tr("The password cannot be empty"));
         return;
     }
```
Comment (deepbluev7) [inline src/UserSettingsPage.cpp:2142]: Can you move this check up to behind the first dialog? There is no need to ask the user for a second password, if the first one is invalid.

File: src/UserSettingsPage.cpp [hunk 3]
```diff
@@ -2127,11 +2127,30 @@ UserSettingsModel::exportSessionKeys()
...
 
+    if (repeatedPassword.isEmpty()) {
+        QMessageBox::warning(nullptr, tr("Error"), tr("Please repeat the password"));
+        return;
+    }
```
Comment (deepbluev7) [inline src/UserSettingsPage.cpp:2147]: This check is unnecessary, if the password was already tested for being empty and password and repeatedPassword have to be the same. You can remove it.

**Review by deepbluev7 — CHANGES_REQUESTED** [review]
Most looks fine, but I think there are some small behavioural things, that could be improved. Thank you for the contribution!

**Comment by Integral-Tech** [comment]
@deepbluev7 Solved :)

**Comment by deepbluev7** [comment]
Sorry, but I don't know what that command means.

Thanks for asking 🥰

**Comment by Integral-Tech** [comment]
@deepbluev7 The pipeline on GitLab seems to failed :(

**Comment by deepbluev7** [comment]
Sorry, but I don't know what that command means.

Thanks for asking 🥰

**Comment by deepbluev7** [comment]
Seems to work now and looks pretty good,thank you!

**Pull request merged**